autoscaler/controllers/datacollection/configmap.go

```diff
@@ -151,7 +151,17 @@ func calculateConfigMapData(nodeCG *odigosv1.CollectorsGroup, sources *odigosv1.
 			"action": "upsert",
 		}},
 	}
-	processorsCfg["resourcedetection"] = config.GenericMap{"detectors": []string{"ec2", "gcp", "azure"}}
+	resourceDetectionProcessor := config.GenericMap{
+		"detectors": []string{"ec2", "azure"},
+		"timeout":   "2s",
+	}
+	// This is a workaround to avoid adding the gcp detector if not running on a gke environment
+	// once https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/issues/1026 is resolved, we can always put the gcp detector
+	if commonconf.ControllerConfig.OnGKE {
+		resourceDetectionProcessor["detectors"] = append(resourceDetectionProcessor["detectors"].([]string), "gcp")
```

Comment:
I know this cant happen but maybe check that "detectors" exists before doing a cast? this has the potential to panic